### ISSUE TITLE
Duplicate call to $this->getLinkField();

### ISSUE THREAD

**Issue by will-b** [magento/magento2#9017]
There is a duplicate call to the getLinkField method inside Magento/Catalog/Model/ResourceModel/Category.php
in the getChildren method

```
$linkField = $this->getLinkField();
$attributeId = $this->getIsActiveAttributeId();
$backendTable = $this->getTable([$this->getEntityTablePrefix(), 'int']);
$connection = $this->getConnection();
$checkSql = $connection->getCheckSql('c.value_id > 0', 'c.value', 'd.value');
$linkField = $this->getLinkField();
```

### Preconditions

1. Magento version 2.1.3
2. Or Magento develop branch

### Steps to reproduce

1. Inspect code of Magento/Catalog/Model/ResourceModel/Category.php

### Expected result

1. Expect $this->getLinkField() not to be called more than required

### Actual result

1. $this->getLinkField() is unnecessarily called a second time

**Comment by davidalger** [comment]
@will-b Mind opening a PR with the suggested fix? I agree from the code you noted above, there is a duplicate call there.

**Comment by will-b** [comment]
@davidalger created PR

**Comment by davidalger** [comment]
@will-b Super, thanks! I'll close this out once the PR is merged into the mainline.

**Comment by magento-team** [comment]
Internal ticket to track issue progress: MAGETWO-66959

**Comment by magento-engcom-team** [comment]
@will-b, thank you for your report.
The issue is already fixed in 2.2.0
